crates/gitbutler-core/src/virtual_branches/remote.rs

```diff
@@ -122,34 +126,57 @@ pub fn branch_to_remote_branch(branch: &git::Branch) -> Result<Option<RemoteBran
             return Ok(None);
         }
     };
-    let name = git::Refname::try_from(branch).context("could not get branch name");
-    match name {
-        Ok(name) => branch
-            .target()
-            .map(|sha| {
-                Ok(RemoteBranch {
-                    sha,
-                    upstream: if let git::Refname::Local(local_name) = &name {
-                        local_name.remote().cloned()
-                    } else {
-                        None
-                    },
-                    name,
-                    last_commit_timestamp_ms: commit
-                        .time()
-                        .seconds()
-                        .try_into()
-                        .map(|t: u128| t * 1000)
-                        .ok(),
-                    last_commit_author: commit
-                        .author()
-                        .name()
-                        .map(std::string::ToString::to_string),
-                })
-            })
-            .transpose(),
-        Err(_) => Ok(None),
-    }
+
+    let name = match git::Refname::try_from(branch) {
+        Ok(name) => name,
+        Err(err) => {
+            tracing::warn!(
+                ?err,
+                "ignoring branch {:?} as refname could not be parsed",
+                branch.name()
+            );
+
+            return Ok(None);
+        }
+    };
+
+    let Some(sha) = branch.target() else {
+        tracing::warn!(
+            "ignoring branch {:?} as target could not be found",
+            branch.name()
+        );
+        return Ok(None);
+    };
+
+    println!("{:?}", name);
```

Comment:
It's probably better use `dbg!()` for this, and if not (because of the added formatting) `eprintln!()` can be used to get text to `stderr` without buffering.

Reply:
That is super neat!

Reply:
One thing that can be confusing is the `dbg!()` moves its arguments, as opposed to `print/eprint` which reference their arguments. This is so that you can just wrap everything in `dbg!()` and it will work like it did before.

So instead of separately adding a line that says `dbg!(foo)`, you'd want to do `mycall(dbg!(foo))` in place and it will act normally.

Reply:
Yes! I saw that in the documentation. Super handy. I actually ended up using it this evening to peek inside some expressions.